### PR TITLE
S28 3223: Custom Health Indicator (Add checks for MK storage account connections)

### PR DESCRIPTION
**Pull request by lucas-phillips28**
### JIRA ticket(s)

- S28-3223

> [!IMPORTANT]
> @lucas-phillips28 Before merge need to add `api-MK-SUBSCRIPTION-ID` and `api-MK-CUSTOMER-ID` secrets to key vault in prod + demo

### Change description
- Add custom health indicator which checks storage account connections for MediaKind.
- Add error logs when MediaKind is not connected to storage accounts corresponding to env
- Add details to health endpoint response which shows connection status for each storage account

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Call the /health endpoint and see details under preApi in response object. `{"preApi": { "mediakindConnections": { "preingestsastg": true, "prefinalsastg": true }}, ...}`

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/preapi/actuator/PreApiHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        return Health.up().withDetail("mediakindConnections", "unknown error").build();
+    }
+
+    private Map<String, Boolean> checkMediaKindConnections() {
+        var storageAccounts = mediaKindAccountsClient.getStorageAccounts();
+        if (storageAccounts.getItems().isEmpty()) {
+            // maybe don't do this for now (will get this always in prod)
+            log.error("MediaKind does not have any storage account connections");
```
Comment (LaurenceRawlings) [inline src/main/java/uk/gov/hmcts/reform/preapi/actuator/PreApiHealthIndicator.java:57]: Does this need to be removed?

Comment (lucas-phillips28) [reply src/main/java/uk/gov/hmcts/reform/preapi/actuator/PreApiHealthIndicator.java:57]: I forgot about this. It should be fine to keep in (the code wont run in prod). I'll remove the comment :)

**Review by jasonpaige — APPROVED** [review]

**Review by cristianemarsilio — APPROVED** [review]
This has been annually tested - details in [S28-3223](https://tools.hmcts.net/jira/browse/S28-3223)

**Comment by lucas-phillips28** [comment]
Added secrets to demo kv

**Pull request merged**